From 5501fcb4c9c154e6454ac0f8e0da0d2a348f0d5e Mon Sep 17 00:00:00 2001
From: zrj <rimvydas.jasinskas@gmail.com>
Date: Wed, 11 Sep 2019 14:36:48 +0300
Subject: [PATCH 4/6] libc: Remove a single use of vm_offset_t type.

 Use (uintptr_t) for casting (void *) like it is already used elsewhere
 for consistency.

 v2: use (uintptr_t) instead of (u_long) as suggested by swildner
---
 lib/libc/sysvipc/shm.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/libc/sysvipc/shm.c b/lib/libc/sysvipc/shm.c
index a90b008a72..2adf107b69 100644
--- a/lib/libc/sysvipc/shm.c
+++ b/lib/libc/sysvipc/shm.c
@@ -33,6 +33,7 @@
 #include <sys/queue.h>
 #include <sys/mman.h>
 #include <sys/shm.h>
+#include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <errno.h>
@@ -244,8 +245,8 @@ sysvipc_shmat(int shmid, const void *shmaddr, int shmflg)
 	flags = MAP_SHARED;
 	if (shmaddr) {
 		if (shmflg & SHM_RND) {
-			addr = (void *)((vm_offset_t)shmaddr & ~(SHMLBA-1));
-		} else if (((vm_offset_t)shmaddr & (SHMLBA-1)) == 0) {
+			addr = (void *)((uintptr_t)shmaddr & ~(SHMLBA-1));
+		} else if (((uintptr_t)shmaddr & (SHMLBA-1)) == 0) {
 			addr = __DECONST(void *, shmaddr);
 		} else {
 			errno = EINVAL;
-- 
2.23.0

